ckcp/gitops.sh

```diff
@@ -153,97 +159,155 @@ install_openshift_gitops() {
+      - op: replace
+        path: /spec/template/spec/containers/0/env/0/value
+        description: This value refers to the hostAddress defined in the Ingress.
+        value: $ckcp_route 
+  - target:
+      kind: Certificate
+      name: kcp
+    patch: |-
+      - op: add
+        path: /spec/dnsNames/-
+        description: This value refers to the hostAddress defined in the Ingress.
+        value: $ckcp_route " >> $ckcp_temp_dir/kustomization.yaml
+    fi
+
+  kustomize build $ckcp_temp_dir | kubectl apply -f - --wait >/dev/null
+  # Check if ckcp pod status is Ready
+  while [[ $(kubectl -n $ns  get pods -l=app=kcp-in-a-pod -o 'jsonpath={.items[0].status.conditions[?(@.type=="Ready")].status}') != "True" ]]; do 
+    echo -n "."
+    sleep 5
+  done  
+
+  rm -rf $ckcp_temp_dir
 
   #############################################################################
   # Post install
   #############################################################################
   # Copy the kubeconfig of kcp from inside the pod onto the local filesystem
-  podname="$(oc get pods --ignore-not-found -n "$ns" -l=app=kcp-in-a-pod -o jsonpath='{.items[0].metadata.name}')"
-  oc cp "$APP/$podname:/workspace/.kcp/admin.kubeconfig" "$KUBECONFIG_KCP" >/dev/null
+  podname="$(kubectl get pods --ignore-not-found -n "$ns" -l=app=kcp-in-a-pod -o jsonpath='{.items[0].metadata.name}')"
+  kubectl cp "$APP/$podname:etc/kcp/config/admin.kubeconfig" "$KUBECONFIG_KCP" >/dev/null
 
   # Check if external ip is assigned and replace kcp's external IP in the kubeconfig file
   echo -n "  - Route: "
-  if grep -q "localhost" "$KUBECONFIG_KCP"; then
-    local route
-    route="$(oc get route ckcp -n "$APP" -o jsonpath='{.spec.host}')"
-    yq e -i "(.clusters[].cluster.server) |= sub(\"localhost:6443\", \"$route:443\")" "$KUBECONFIG_KCP"
+  if grep -q "ckcp-ckcp.apps.domain.org" "$KUBECONFIG_KCP"; then
+    yq e -i "(.clusters[].cluster.server) |= sub(\"ckcp-ckcp.apps.domain.org:6443\", \"$ckcp_route:443\")" "$KUBECONFIG_KCP"
   fi
   echo "OK"
 
-  # Make sure access to kcp-in-a-pod is good
-  echo -n "  - KCP api-server: "
-  KUBECONFIG="$KUBECONFIG_KCP" oc config set-cluster admin --insecure-skip-tls-verify=true >/dev/null
-  until KUBECONFIG="$KUBECONFIG_KCP" oc api-resources >/dev/null 2>&1; do
-    echo -n "."
-    sleep 2
-  done
-  echo "OK"
-
   # Register the host cluster to KCP
-  echo -n "  - Workloadcluster pipeline-cluster registration: "
-  if ! KUBECONFIG="$KUBECONFIG_KCP" oc get cluster local >/dev/null 2>&1; then
-    (
-      curl --fail --silent https://raw.githubusercontent.com/kcp-dev/kcp/948dbe9565cc7da439c698875ca1fa78350c4530/contrib/examples/cluster.yaml
-      sed -e 's/^/    /' "$KUBECONFIG"
-    ) >"$kube_dir/cluster.yaml"
-    KUBECONFIG="$KUBECONFIG_KCP" oc apply -f "$kube_dir/cluster.yaml" >/dev/null
-    rm "$kube_dir/cluster.yaml"
+  echo -n "  - Workspace: "
+  # It's not allowed to create WorkloadCluster resource in a non-private workspace
```

Comment:
["universal" ](https://github.com/kcp-dev/kcp/blob/main/config/organization/clusterworkspacetype-universal.yaml)is the current denomination for default user workspaces. If I am not mistaken other workspace types may allow the creation of application resources but not root and organization workspaces. "private" is not what they are being called.

Reply:
fixed using `a non-universal workspace`